web/modules/custom/joinup_federation/src/Plugin/pipeline/Step/UserSelectionFilter.php

```diff
@@ -137,25 +136,19 @@ public function execute() {
     }
 
     // Build a list of all whitelisted entities.
-    $this->buildWhitelist('solution', $selected_solution_ids);
-
-    // Get all imported entity IDs by running a SPARQL query.
-    /** @var \EasyRdf\Sparql\Result $results */
-    $results = $this->sparql->query("SELECT DISTINCT(?entityId) WHERE { GRAPH <{$this->getGraphUri('sink')}> { ?entityId ?p ?o . } }");
-    $all_imported_ids = array_map(function (\stdClass $item): string {
-      return $item->entityId->getUri();
-    }, $results->getArrayCopy());
-
-    // Remove the blacklisted entities, if any.
-    if ($blacklist = array_values(array_diff($all_imported_ids, $this->whitelist))) {
-      $blacklist_ids = SparqlArg::serializeUris($blacklist, ' ');
-      $this->sparql->query("WITH <{$this->getGraphUri('sink')}> DELETE { ?entity_id ?p ?o } WHERE { ?entity_id ?p ?o . VALUES ?entity_id { {$blacklist_ids} } }");
+    $this->buildWhitelist($selected_solution_ids);
+    $all_imported_ids = $this->getAllIncomingIds();
+
+    // Remove the not selected entities, if any.
+    if ($not_selected = array_values(array_diff($all_imported_ids, $this->whitelist))) {
+      $not_selected_ids = SparqlArg::serializeUris($not_selected, ' ');
+      $this->sparql->query("WITH <{$this->getGraphUri('sink')}> DELETE { ?entity_id ?p ?o } WHERE { ?entity_id ?p ?o . VALUES ?entity_id { {$not_selected_ids} } }");
```

Comment:
Actually, I didn't touch this line. This was how it was before. I think it was like that because this is the form submission of the UserSeleftionFilter and if there are too many solutions it would be slow to load all of them and then delete them (the `::deleteFromGraph` is receiving full entities), while this is blazing fast.